docs/running-on-yarn.md

```diff
@@ -82,6 +82,19 @@ In `cluster` mode, the driver runs on a different machine than the client, so `S
 
 Running Spark on YARN requires a binary distribution of Spark which is built with YARN support.
 Binary distributions can be downloaded from the [downloads page](https://spark.apache.org/downloads.html) of the project website.
+There are two variants of Spark binary distributions you can download. One is pre-built with a certain
+version of Apache Hadoop; this Spark distribution contains built-in Hadoop runtime, so we call it `with-hadoop` Spark
+distribution. The other one is pre-built with user-provided Hadoop; since this Spark distribution
+doesn't contain a built-in Hadoop runtime, it's smaller, but users have to provide a Hadoop installation separately.
+We call this variant `no-hadoop` Spark distribution. For `with-hadoop` Spark distribution, since
+it contains a built-in Hadoop runtime already, by default, when a job is submitted to Hadoop Yarn cluster, to prevent jar conflict, it will not
+populate Yarn's classpath into Spark. To override this behavior, you can set <code>spark.yarn.populateHadoopClasspath=true</code>.
+For `no-hadoop` Spark distribution, Spark will populate Yarn's classpath by default in order to get Hadoop runtime. Note that some features such
+as Hive support are not available in `no-hadoop` Spark distribution. For `with-hadoop` Spark distribution,
```

Comment:
is this true, hive will not work?  there is a separate hive-provided flag.

Reply:
Maybe I'm wrong, but I got the impression from @dongjoon-hyun that no-hadoop Spark distribution doesn't support Hive.

Reply:
? This is added at the last commit. I didn't say like that, @dbtsai . :)

Reply:
You can make `no-hadoop` distribution into `hadoop` distribution if you copy the exclude jars like `with-hadoop` distribution. If you make have `with-hadoop` distribution back, there is no difference, isn't it?

Reply:
Okay, I'm getting old now :) Just somehow remember you told me this, and that was why our internal `no-hadoop` build can not support hive. I removed those lines. Thanks.